### PR TITLE
 Improve search by content by matching against filename instead of full path

### PR DESCRIPTION
**Pull request by Arignir**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -458,15 +458,33 @@ Upload a package from its NPF (`.nest`) form. Its name, category and version are
 
 **Note**: The package must be in the NPF format. The best way to generate a package under this format is to use [`nbuild`](https://github.com/raven-os/nbuild).
 
+**Note**: In order to ensure a fast response time to this request even if the NPF is huge, the NPF is parsed and made publicly visible asynchronously.
+Only a quick check of the package is done synchronously in order to find most erroneous NPF and yield the content of the reponse body.
```
Comment (doom) [inline API.md]:
```suggestion
Only a quick check of the package is done synchronously in order to find most erroneous NPFs and yield the content of the response body.
```

File: src/api/upload.rs [hunk 2]
```diff
@@ -7,16 +7,18 @@ use failure::Error;
 use libnest::package::NPFExplorer;
 use rocket::http::Status;
 use rocket::{Data, State};
+use rocket_contrib::json;
+use rocket_contrib::json::{JsonValue};
```
Comment (doom) [inline src/api/upload.rs]: How about `use rocket_contrib::json::{self, JsonValue};` :p

Comment (Arignir) [reply src/api/upload.rs]: You'd be surprise to discover that... it doesn't work :D  (I hoped you would ask that :D)

Here `rocket_contrib::json` is a macro, not a module, so using `rocket_contrib::json::self` wouldn't refer to the same item.

(Yes, it sucks that Rust allows two items in the same scope with the same name in the first place, I agree, but here we are)

Comment (doom) [reply src/api/upload.rs]: I think "Bleh :nauseated_face:" is the appropriate reaction to this

**Pull request merged**